### PR TITLE
BUG: downcast = 'unsigend' on 0 would would not downcast to unsigned.

### LINKED ISSUE #14401: BUG: pd.to_numeric(..., downcast='unsigned') should accept 0
with 0.19.0.

```
In [28]: pd.to_numeric(pd.Series([1,2,3]), downcast='unsigned')
Out[28]: 
0    1
1    2
2    3
dtype: uint8

In [29]: pd.to_numeric(pd.Series([0,1,2,3]), downcast='unsigned')
Out[29]: 
0    0
1    1
2    2
3    3
dtype: int64
```

```
In [30]: np.iinfo(np.uint8).min
Out[30]: 0
```

I think 0 is a legit uint8, so [29] should be uint8.

### PR DESCRIPTION
**Pull request by verhalenn**
Continuation of #14412
- [ ] closes #14401
- [ ] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

Once again I had some very bizarre issues with git. Hope this does it this time.

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
@verhalenn For future reference, it is normally never needed to open a new PR to fix git issues. You can just clean it up locally and force push.

BTW, you have some failing tests.

**Comment by verhalenn** [comment]
No when I forced pushed it it closed the pull request for some reason. Usually that would happen if I were to delete the branch and then force push but that wasn't the case. I'll look into the error.

File: pandas/tools/tests/test_util.py [hunk 1]
```diff
@@ -401,6 +401,32 @@ def test_downcast(self):
             res = pd.to_numeric(data, downcast=downcast)
             tm.assert_numpy_array_equal(res, expected)
 
+        # check that the smallest and largest values in each integer type pass to each type.
```
Comment (jreback) [inline pandas/tools/tests/test_util.py]: can you make this into a separate test function

add the issue number as a comment

ping on green

File: doc/source/whatsnew/v0.19.1.txt [hunk 2]
```diff
@@ -56,6 +56,5 @@ Bug Fixes
 - Bug in ``MultiIndex.set_levels`` where illegal level values were still set after raising an error (:issue:`13754`)
 - Bug in ``DataFrame.to_json`` where ``lines=True`` and a value contained a ``}`` character (:issue:`14391`)
 - Bug in ``df.groupby`` causing an ``AttributeError`` when grouping a single index frame by a column and the index level (:issue`14327`)
-
 - Bug in ``pd.pivot_table`` may raise ``TypeError`` or ``ValueError`` when ``index`` or ``columns``
-  is not scalar and ``values`` is not specified (:issue:`14380`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.19.1.txt]: you eliminated this line, pls restore it (its another issue)

**Comment by jreback** [comment]
thanks!

**Comment by gfyoung** [comment]
@jreback : I think this commit is breaking `master` right now.

**Comment by jreback** [comment]
reverted: https://github.com/pandas-dev/pandas/commit/d1d75d7fcb6b5d090cf96d328d0136c3959a82b8


@verhalenn if you can re-submit this PR would be great.

**Comment by verhalenn** [comment]
First thing tomorrow morning. I was kind of surprised you closed it when you did. The lint should be an easy fix but the downcast test will take a little extra time.

**Pull request closed without merging**